pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+        UnknownEraReward,
+        AlreadyClaimedInThisEra,
+        MaintenanceModeNotChanged,
+        RewardDistributionConfigInconsistent,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+            // As long as pallet is disabled, we shouldn't allow any storage modifications.
+            // This means we might prolong an era but it's acceptable.
+            // Runtime upgrade should be timed so we ensure that we complete it before
+            // a new era is triggered. This code is just a safety net to ensure nothing is broken
+            // if we fail to do that.
+            if PalletDisabled::<T>::get() {
+                return T::DbWeight::get().reads(1);
+            }
+
+            let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
+            let previous_era = Self::current_era();
+            let next_era_starting_block = Self::next_era_starting_block();
+
+            // Value is compared to 1 since genesis block is ignored
+            if now >= next_era_starting_block || force_new_era || previous_era.is_zero() {
+                let blocks_per_era = T::BlockPerEra::get();
+                let next_era = previous_era + 1;
+                CurrentEra::<T>::put(next_era);
+
+                NextEraStartingBlock::<T>::put(now + blocks_per_era);
+
+                let reward = BlockRewardAccumulator::<T>::take();
+                Self::reward_balance_snapshot(previous_era, reward);
+                let consumed_weight = Self::rotate_staking_info(previous_era);
+
+                if force_new_era {
+                    ForceEra::<T>::put(Forcing::NotForcing);
+                }
+
+                Self::deposit_event(Event::<T>::NewCreatorStakingEra { number: next_era });
+
+                consumed_weight + T::DbWeight::get().reads_writes(5, 3)
+            } else {
+                T::DbWeight::get().reads(4)
+            }
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(3, 1))]
+        pub fn force_register_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
+
+            ensure!(
+                !RegisteredCreators::<T>::contains_key(space_id),
+                Error::<T>::AlreadyUsedCreatorSpace,
+            );
+
+            let space_owner = T::SpacesInterface::get_space_owner(space_id)?;
+            T::Currency::reserve(&space_owner, T::RegistrationDeposit::get())?;
+
+            RegisteredCreators::<T>::insert(space_id, CreatorInfo::new(space_owner.clone()));
+
+            Self::deposit_event(Event::<T>::CreatorRegistered { who: space_owner, space_id });
+
+            Ok(().into())
```

Comment:
```suggestion
            Ok(Pays::No.into())
```

Reply:
@F3Joule looks like you skipped the suggestion